packages/sanity/src/core/form/studio/tree-editing/__tests__/treeEditingSearch.test.ts

```diff
@@ -0,0 +1,109 @@
+import {describe, expect, test} from '@jest/globals'
+
+import {treeEditingSearch} from '../components/search/utils'
+import {type TreeEditingMenuItem} from '../types'
+
+const ITEMS: TreeEditingMenuItem[] = [
+  {
+    path: ['path-1'],
+    title: 'Item 1',
+  },
+  {
+    path: ['path-2'],
+    title: 'Item 2',
+    children: [
+      {
+        path: ['path-2', 'child-1'],
+        title: 'Child 1',
+        children: [
+          {
+            path: ['path-2', 'child-1', 'grandchild-1'],
+            title: 'Grandchild 1',
+          },
+          {
+            path: ['path-2', 'child-1', 'grandchild-2'],
+            title: 'Grandchild 2',
+          },
+        ],
+      },
+    ],
+  },
+  {
+    path: ['path-3'],
+    title: 'Item 3',
+  },
+]
+
+describe('tree-editing: search', () => {
+  test('should return the correct result when searching', () => {
+    // Case 1
+    const result1 = treeEditingSearch(ITEMS, 'Item 1')
+
+    const expected1: TreeEditingMenuItem[] = [
+      {
+        path: ['path-1'],
+        title: 'Item 1',
+      },
+    ]
+
+    // Case 2
+    const result2 = treeEditingSearch(ITEMS, 'Item')
+
+    const expected2: TreeEditingMenuItem[] = [
+      {
+        path: ['path-1'],
+        title: 'Item 1',
+      },
+      {
+        path: ['path-2'],
+        title: 'Item 2',
+      },
+      {
+        path: ['path-3'],
+        title: 'Item 3',
+      },
+    ]
+
+    // Case 3
```

Comment:
Instead of adding comments, I split the test cases into individual tests, each with its own description.

https://github.com/sanity-io/sanity/pull/6804/commits/527b2b879972558989bed0ee51248675d367dcdd

Reply:
Marvellous 😚 👌